compiler/rustc_codegen_llvm/src/gotoc/statement.rs

```diff
@@ -106,22 +106,18 @@ impl<'tcx> GotocCtx<'tcx> {
             let le = self.codegen_place(location).goto_expr;
             hk.handle(self, instance, vec![le], None, Some(*target), None)
         } else {
-            let drop_implementation = match instance.def {
-                InstanceDef::Virtual(..) => self
-                    .codegen_unimplemented(
-                        "drop_in_place for dynamic argument",
-                        Type::Empty,
-                        Location::none(),
-                        "https://github.com/model-checking/rmc/issues/11",
-                    )
-                    .as_stmt(Location::none()),
-                InstanceDef::DropGlue(..) => Stmt::skip(Location::none()),
-                _ => {
-                    let func = self.codegen_func_expr(instance, None);
-                    func.call(vec![self.codegen_place(location).goto_expr.address_of()])
-                        .as_stmt(Location::none())
-                }
-            };
+            let drop_implementation;
+            // For empty DropGlue function invocation, we ignore it
+            if let InstanceDef::DropGlue(_, None) = instance.def {
+                drop_implementation = Stmt::skip(Location::none());
+            } else if let ty::Dynamic(..) = loc_ty.kind() {
+                drop_implementation = Stmt::skip(Location::none());
+            } else {
```

Comment:
get rid of explicit catch all, be explicit about what types we handle, and then panic on the rest